src/java/com/reucon/openfire/plugin/archive/impl/PaginatedMessageDatabaseQuery.java

```diff
@@ -158,6 +159,7 @@ public String toString()
                 final ArchivedMessage archivedMessage = JdbcPersistenceManager.extractMessage(owner, rs);
                 archivedMessages.add(archivedMessage);
             }
+            archivedMessages.sort(Comparator.comparing(ArchivedMessage::getTime));
```

Comment:
Can this be improved by reversing the order of the list, instead of re-ordering it by a property? This probably will have the same result with the current code, but it would keep the responsibility of defining order in one place.

Also, changelog. :)

Reply:
The results could be in either order when they come out. I'll look at selectively reversing. That would be more performant. And 🤦‍♂️, of course I'll add a change log entry.

Reply:
Would it be as simple as evaluating `isPagingBackwards` - or am I missing bits then?

Reply:
Turns out: yes. It was exactly that simple.